### PR TITLE
Bugfix/stderr

### LINKED ISSUE #5: Errors are not being captured
If you run a command that does not exist it's not being printed to the history file.

### PR DESCRIPTION
**Pull request by thiagonache**
The stderr message was not being captured in the history log or printed to the user neither.

### CODE REVIEW

**Comment by thiagonache** [comment]
Handle #5

File: history.go [hunk 1]
```diff
@@ -8,31 +8,31 @@ import (
...
 
 // NewRecorder instantiate a new Recorder object and returns a pointer to it.
 func NewRecorder() (*Recorder, error) {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
 	return &Recorder{
-		Ctx:        context.Background(),
-		Path:       "history.log",
-		Permission: 0664,
-		Signals:    []os.Signal{os.Interrupt},
+		Ctx:        ctx,
```
Comment (bitfield) [inline history.go]: Maybe `Context` is clearer here?

File: history.go [hunk 2]
```diff
@@ -55,7 +55,9 @@ func (r *Recorder) EnsureHistoryFileOpen() error {
 func (r *Recorder) Session() {
 	err := r.EnsureHistoryFileOpen()
 	if err != nil {
-		fmt.Fprint(r.Stdout, err)
+		r.Stop()
+		fmt.Fprintln(r.Stdout, err)
```
Comment (bitfield) [inline history.go]: This should go to standard error, shouldn't it?

File: history.go [hunk 3]
```diff
@@ -101,11 +102,21 @@ func (r *Recorder) Execute(command string) error {
 	return nil
 }
 
+// SetPath takes a string and set history log file path
+func (r *Recorder) SetPath(path string) {
```
Comment (bitfield) [inline history.go:128]: You could use functional options for this, couldn't you, as per Code Club?

File: history_test.go [hunk 4]
```diff
@@ -16,28 +17,57 @@ type testWriteCloser struct {
...
-	}
-	r.Stdout = &output
-	err = r.Execute(command)
-	if err != nil {
-		t.Fatal(err)
+	testCases := []struct {
+		command     string
+		desc        string
+		errExpected bool
+		output      bytes.Buffer
```
Comment (bitfield) [inline history_test.go]: I don't see where `output` is set in any of the test cases?

File: history_test.go [hunk 5]
```diff
@@ -16,28 +17,57 @@ type testWriteCloser struct {
...
+			want:        "exec: \"abc\": executable file not found in $PATH",
+			errExpected: true,
+		},
 	}
-	wantOutput := "testing\n"
-	if !cmp.Equal(wantOutput, output.String()) {
-		t.Error(cmp.Diff(wantOutput, output.String()))
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			r, err := history.NewRecorder()
```
Comment (bitfield) [inline history_test.go]: Do we need a new Recorder for every test case, or could we just execute all the test commands on a single Recorder?

**Review by bitfield — COMMENTED** [review]
This is looking great!

File: history.go (hunk 1 above)
Comment (thiagonache) [reply history.go]: I think I tried but Context was conflicting with the context.Context. I'll try it again. Maybe the error was somewhere else.

File: history.go (hunk 2 above)
Comment (thiagonache) [reply history.go]: good catch! thx.

File: history.go (hunk 3 above)
Comment (thiagonache) [reply history.go:128]: Yeah. Thank you!

File: history_test.go (hunk 5 above)
Comment (thiagonache) [reply history_test.go]: One should be good, thx.

File: history.go (hunk 1 above)
Comment (thiagonache) [reply history.go]: updated

File: history_test.go (hunk 4 above)
Comment (thiagonache) [reply history_test.go]: line 51: r.Stdout = &tC.output

File: history.go (hunk 3 above)
Comment (thiagonache) [reply history.go:128]: after second thought, I want to keep the variable private and have methods to deal with it. Is there anything wrong?

**Pull request merged**